hypothesis/hypo_vector.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+import math
+
+class segment_hypo(object):
+    '''
+    create hypo using individual segments and retrieve matrix that contains expected photons in each cell in spherical coordinate system with dom at origin.
+    binnnings and location of the dom must be set
+    '''
+    def __init__(self, t_v, x_v, y_v, z_v, theta_v, phi_v, trck_energy, cscd_energy, time_increment=1.):
+        '''
+        provide vertex and track information
+        t_v : time (ns)
+        x_v, y_v, z_v : vertex position (m)
+        theta : zenith (rad)
+        phi : azimuth (rad)
+        trck_energy : track energy (GeV)
+        cscd_energy : cascade energy (GeV)
+        time_increment : if using constant time increments, it is the length of time between photon dumps in ns
+        '''     
+        #assign vertex
+        self.t = t_v * 1e-9
+        self.x = x_v
+        self.y = y_v
+        self.z = z_v
+        self.theta_v = theta_v
+        self.phi_v = phi_v
+        #declaring constants 
+        self.time_increment = time_increment * 1e-9
+        self.speed_of_light = 2.99e8
+        self.segment_length = self.time_increment * self.speed_of_light
+        #calculate frequently used values
+        self.sin_theta_v = math.sin(self.theta_v)
+        self.cos_theta_v = math.cos(self.theta_v)
+        self.sin_phi_v = math.sin(self.phi_v)
+        self.cos_phi_v = math.cos(self.phi_v)
+        self.speed_x = self.speed_of_light * self.sin_theta_v * self.cos_phi_v
+        self.speed_y = self.speed_of_light * self.sin_theta_v * self.sin_phi_v
+        self.speed_z = self.speed_of_light * self.cos_theta_v
+        #convert track energy to length
+        self.trck_length = 15. / 3.3 * trck_energy
+        #precalculated (nphotons.py) to avoid icetray
+        self.photons_per_meter = 2451.4544553
+        self.cscd_photons = 12805.3383311 * cscd_energy
+        self.trck_photons = self.trck_length * self.photons_per_meter
+        self.tot_photons = self.cscd_photons + self.trck_photons
+
+    
+    def set_binning(self, n_t_bins, n_r_bins, n_theta_bins, n_phi_bins, t_max, r_max, t_min=0, r_min=0):
+        '''
```

Comment:
add some more details, is it assumed that the binning in r is quadratic for example, and theta is in cos(theta)?